cmd/pinshare-tray/config.go

```diff
@@ -0,0 +1,73 @@
+package main
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+)
+
+// Default ports (must match pinsharesvc defaults)
+const (
+	defaultIPFSAPIPort     = 5001
+	defaultPinShareAPIPort = 9090
+)
+
+// TrayConfig holds configuration values needed by the tray application
+type TrayConfig struct {
+	IPFSAPIPort     int `json:"ipfs_api_port"`
+	PinShareAPIPort int `json:"pinshare_api_port"`
+}
+
+// Global config instance
+var appConfig *TrayConfig
+
+// loadConfig loads configuration from config.json
+// Falls back to defaults if config file doesn't exist or can't be read
+func loadConfig() *TrayConfig {
+	config := &TrayConfig{
+		IPFSAPIPort:     defaultIPFSAPIPort,
+		PinShareAPIPort: defaultPinShareAPIPort,
+	}
+
+	programData := os.Getenv("PROGRAMDATA")
+	if programData == "" {
+		programData = `C:\ProgramData`
+	}
+
+	configPath := filepath.Join(programData, "PinShare", "config.json")
+
+	data, err := os.ReadFile(configPath)
+	if err != nil {
+		// Config file doesn't exist yet, use defaults
+		return config
```

Comment:
If my understanding is correct that in this case no config file exists, I think we should create one from the defaults.

Reply:
The config file is created by the service during \ (see \). The tray app is a control panel for the service and reads the config but shouldn't create it - if no config exists, the service likely isn't installed. Creating a config from the tray could cause issues if the service later tries to initialize with different defaults. The tray gracefully falls back to defaults for display purposes without persisting them.